File: server/src/models/bank_account.model.js
// src/models/bank_account.model.js
class BankAccount {
  constructor({
    bank_account_id,
    user_id,
    hotel_id,
    holder_name,
    account_number,
    bank_name,
    branch_name,
    is_default,
    status,
    created_at,
    updated_at,
  }) {
    this.bankAccountId = bank_account_id;
    this.userId = user_id;
    this.hotelId = hotel_id || null;
    this.holderName = holder_name;
    this.accountNumber = account_number;
    this.bankName = bank_name;
    this.branchName = branch_name || null;
    this.isDefault = is_default || false;
    this.status = status || 'active';
    this.createdAt = created_at;
    this.updatedAt = updated_at;
  }

  static fromDB(row) { 
    return new BankAccount(row); 
  }

  toJSON() {
    return {
      bankAccountId: this.bankAccountId,
      userId: this.userId,
      hotelId: this.hotelId,
      holderName: this.holderName,
      accountNumber: this.accountNumber,
      bankName: this.bankName,
      branchName: this.branchName,
      isDefault: this.isDefault,
      status: this.status,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt,
    };
  }

  toDBObject() {
    return {
      bank_account_id: this.bankAccountId,
      user_id: this.userId,
      hotel_id: this.hotelId,
      holder_name: this.holderName,
      account_number: this.accountNumber,
      bank_name: this.bankName,
      branch_name: this.branchName,
      is_default: this.isDefault,
      status: this.status,
      created_at: this.createdAt,
      updated_at: this.updatedAt,
    };
  }

  // Validation methods
  static validateStatus(status) {
    const validStatuses = ['active', 'inactive'];
    return validStatuses.includes(status);
  }

  static validateAccountNumber(accountNumber) {
    // Basic validation - account number should be numeric and 6-20 digits
    return /^\d{6,20}$/.test(accountNumber);
  }

  static validateBankName(bankName) {
    // Bank name should not be empty and reasonable length
    return bankName && bankName.trim().length >= 2 && bankName.trim().length <= 100;
  }

  static validateHolderName(holderName) {
    // Holder name should not be empty and reasonable length
    return holderName && holderName.trim().length >= 2 && holderName.trim().length <= 100;
  }
}

module.exports = BankAccount;

File: mobile/lib/classes/bank_account_model.dart
// ============================================
// Bank Account MODEL
// ============================================

class BankAccount {
  final String bankAccountId;
  final String userId;
  final String? hotelId;
  final String holderName;
  final String accountNumber;
  final String bankName;
  final String? branchName;
  final bool isDefault;
  final String status;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  BankAccount({
    required this.bankAccountId,
    required this.userId,
    this.hotelId,
    required this.holderName,
    required this.accountNumber,
    required this.bankName,
    this.branchName,
    required this.isDefault,
    required this.status,
    this.createdAt,
    this.updatedAt,
  });

  factory BankAccount.fromJson(Map<String, dynamic> json) {
    return BankAccount(
      bankAccountId: json['bankAccountId'] ?? json['bank_account_id'] ?? '',
      userId: json['userId'] ?? json['user_id'] ?? '',
      hotelId: json['hotelId'] ?? json['hotel_id'],
      holderName: json['holderName'] ?? json['holder_name'] ?? '',
      accountNumber: json['accountNumber'] ?? json['account_number'] ?? '',
      bankName: json['bankName'] ?? json['bank_name'] ?? '',
      branchName: json['branchName'] ?? json['branch_name'],
      isDefault: json['isDefault'] ?? json['is_default'] ?? false,
      status: json['status'] ?? 'active',
      createdAt: json['createdAt'] != null
          ? DateTime.tryParse(json['createdAt'])
          : (json['created_at'] != null ? DateTime.tryParse(json['created_at']) : null),
      updatedAt: json['updatedAt'] != null
          ? DateTime.tryParse(json['updatedAt'])
          : (json['updated_at'] != null ? DateTime.tryParse(json['updated_at']) : null),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'bankAccountId': bankAccountId,
      'userId': userId,
      'hotelId': hotelId,
      'holderName': holderName,
      'accountNumber': accountNumber,
      'bankName': bankName,
      'branchName': branchName,
      'isDefault': isDefault,
      'status': status,
      'createdAt': createdAt?.toIso8601String(),
      'updatedAt': updatedAt?.toIso8601String(),
    };
  }
}
